apps/dc_tools/odc/apps/dc_tools/_stac.py

```diff
@@ -171,19 +172,29 @@ def _get_stac_bands(
     grids = {}
     accessories = {}
 
+    self_link = _find_self_href(item)
+
     assets = item.get("assets", {})
 
-    def _get_path(asset):
-        path = asset["href"]
+    def _get_path(asset, force_relative=False):
+        path = URL(asset["href"])
         if relative:
-            path = Path(path).name
+            try:
+                path = path.relative_to(URL(self_link).parent)
+            except (ValueError, TypeError):
+                if force_relative:
+                    path = path.name
```

Comment:
Should have at least some comment describing what happens and in what situations. I can only guess from looking? If asset doesn't share same path with the stac item, just use filename as relative path?

Reply:
On it.